### PR TITLE
provider/google: Internal load balancer caching agent.

### PR DESCRIPTION
**Pull request by jtk54**
@duftler please review. Provides the caching agent for ILBs, as well as a bit of refactoring for new types of health checks, load balancers, and load balancing policies.

### CODE REVIEW

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleInternalLoadBalancerCachingAgent.groovy [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+  Collection<Map> pendingOnDemandRequests(ProviderCache providerCache) {
+    return []
+  }
+
+  /**
+   * This is a "simpleton" way of handling on-demand cache requests. Load Balancer mutation (and thus the need for
+   * cache refreshing) is not as common or complex as server group cache refreshes.
+   *
+   * This implementation has the potential for race condition between handle() and loadData(), which may
+   * cause "flapping" in the UI. lwander@ has plans to make an abstract solution for this race condition, so this impl
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleInternalLoadBalancerCachingAgent.groovy]: I've seen this note in earlier PR's as well. Please ensure we have a bug to track that fix.

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleLoadBalancerCachingAgent.groovy [hunk 2]
```diff
@@ -190,16 +191,15 @@ class GoogleLoadBalancerCachingAgent extends AbstractGoogleCachingAgent implemen
...
 
-        if (forwardingRule.target) {
           def forwardingRuleTokens = forwardingRule.target.split("/")
 
           if (forwardingRuleTokens[forwardingRuleTokens.size() - 2] != "targetVpnGateways") {
             def targetPoolName = Utils.getLocalName(forwardingRule.target)
             def targetPoolsCallback = new TargetPoolCallback(googleLoadBalancer: newLoadBalancer,
-                                                             httpHealthChecksRequest: httpHealthChecksRequest,
-                                                             instanceHealthRequest: instanceHealthRequest)
+              httpHealthChecksRequest: httpHealthChecksRequest,
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleLoadBalancerCachingAgent.groovy]: Inadvertent reformatting?

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/config/GoogleInfrastructureProviderConfig.groovy [hunk 3]
```diff
@@ -127,6 +127,11 @@ class GoogleInfrastructureProviderConfig {
                                                                    objectMapper,
                                                                    registry)
         regions.each { String region ->
+          newlyAddedAgents << new GoogleInternalLoadBalancerCachingAgent(googleConfiguration.googleApplicationName(),
+                                                                 credentials,
```
Comment (duftler) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/config/GoogleInfrastructureProviderConfig.groovy]: Indentation is off here.

Comment (jtk54) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/config/GoogleInfrastructureProviderConfig.groovy]: Done.

**Pull request merged**